contrib/turbojpeg/tests/turbojpeg_sapi_test.cc

```diff
@@ -0,0 +1,374 @@
+  0x69, 0x64, 0x20, 0x2a, 0x3e, 0x3e, 0x2c, 0x20, 0x22, 0x62, 0x75, 0x67,
+  0x22, 0x29, 0x3b, 0x0a, 0x20, 0x20, 0x61, 0x62, 0x73, 0x6c, 0x3a, 0x3a,
+  0x53, 0x74, 0x61, 0x74, 0x75, 0x73, 0x4f, 0x72, 0x3c, 0x76, 0x6f, 0x69,
+  0x64, 0x20, 0x2a, 0x3e, 0x20, 0x64, 0x65, 0x63, 0x6f, 0x6d, 0x70, 0x72,
+  0x65, 0x73, 0x73, 0x69, 0x6f, 0x6e, 0x5f, 0x68, 0x61, 0x6e, 0x64, 0x6c,
+  0x65, 0x5f, 0x72, 0x61, 0x77, 0x20, 0x3d, 0x20, 0x61, 0x70, 0x69, 0x5f,
+  0x2d, 0x3e, 0x74, 0x6a, 0x49, 0x6e, 0x69, 0x74, 0x44, 0x65, 0x63, 0x6f,
+  0x6d, 0x70, 0x72, 0x65, 0x73, 0x73, 0x28, 0x29, 0x3b, 0x0a, 0x20, 0x20,
+  0x41, 0x53, 0x53, 0x45, 0x52, 0x54, 0x5f, 0x54, 0x48, 0x41, 0x54, 0x28,
+  0x64, 0x65, 0x63, 0x6f, 0x6d, 0x70, 0x72, 0x65, 0x73, 0x73, 0x69, 0x6f,
+  0x6e, 0x5f, 0x68, 0x61, 0x6e, 0x64, 0x6c, 0x65, 0x5f, 0x72, 0x61, 0x77,
+  0x2c, 0x20, 0x49, 0x73, 0x4f, 0x6b, 0x28, 0x29, 0x29, 0x3b, 0x0a, 0x20,
+  0x20, 0x74, 0x6a, 0x68, 0x61, 0x6e, 0x64, 0x6c, 0x65, 0x20, 0x64, 0x65,
+  0x63, 0x6f, 0x6d, 0x70, 0x72, 0x65, 0x73, 0x73, 0x69, 0x6f, 0x6e, 0x5f,
+  0x68, 0x61, 0x6e, 0x64, 0x6c, 0x65, 0x20, 0x3d, 0x20, 0x64, 0x65, 0x63,
+  0x6f, 0x6d, 0x70, 0x72, 0x65, 0x73, 0x73, 0x69, 0x6f, 0x6e, 0x5f, 0x68,
+  0x61, 0x6e, 0x64, 0x6c, 0x65, 0x5f, 0x72, 0x61, 0x77, 0x2e, 0x76, 0x61,
+  0x6c, 0x75, 0x65, 0x28, 0x29, 0x3b, 0x0a, 0x20, 0x20, 0x41, 0x53, 0x53,
+  0x45, 0x52, 0x54, 0x5f, 0x54, 0x52, 0x55, 0x45, 0x28, 0x64, 0x65, 0x63,
+  0x6f, 0x6d, 0x70, 0x72, 0x65, 0x73, 0x73, 0x69, 0x6f, 0x6e, 0x5f, 0x68,
+  0x61, 0x6e, 0x64, 0x6c, 0x65, 0x20, 0x21, 0x3d, 0x20, 0x6e, 0x75, 0x6c,
+  0x6c, 0x70, 0x74, 0x72, 0x29, 0x3b, 0x0a, 0x7d, 0x0a, 0x7d, 0x0a, 0x00,
+};
+unsigned int image_len = 2412;
+// clang-format on
+
+#ifdef DEBUG_LOGGING
+#define MADE_IT                                                   \
+  do {                                                            \
+    std::cerr << "Reached " << __FILE__ << ':' << __LINE__ << ' ' \
+              << __PRETTY_FUNCTION__ << std::endl;                \
+  } while (0)
+#else
+#define MADE_IT \
+  do {          \
+  } while (0)
+#endif
+
+TEST_F(TurboJpegSapiSandboxTest, Compressor) {
+  static_assert(
+      std::is_same_v<decltype(api_->tjInitCompress()), absl::StatusOr<void*>>,
+      "bug");
+  absl::StatusOr<void*> compression_handle_raw = api_->tjInitCompress();
+  ASSERT_THAT(compression_handle_raw, IsOk());
+  ASSERT_THAT(compression_handle_raw.value(), NotNull());
+  sapi::v::RemotePtr compression_handle{compression_handle_raw.value()};
+
+  sapi::v::Array<unsigned char> array{sizeof image};
+  std::memcpy(array.GetData(), image, sizeof image);
+
+  sapi::v::GenericPtr buffer;
+  unsigned long actual_length;
+  {
+    sapi::v::ULong length{0};
+    auto result = api_->tjCompress2(
+        &compression_handle, array.PtrBefore(), 12, 36, 67, TJPF_RGB,
+        buffer.PtrAfter(), length.PtrBoth(), TJSAMP_444, 10,
+        0 * (TJFLAG_ACCURATEDCT | TJFLAG_PROGRESSIVE | TJFLAG_LIMITSCANS));
```

Comment:
Flags will be zero. If that's on purpose, then why have the other constants here?